runtime/vm/visible.c

```diff
@@ -141,12 +146,34 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 			}
 		} else if (modifiers & J9AccPrivate) {
 			/* Private */
-			if ((sourceClass != destClass)
+			if (sourceClass != destClass) {
 #if defined(J9VM_OPT_VALHALLA_NESTMATES)
-					&& (sourceClass->nestHost != destClass->nestHost)
+				/* loadAndVerifyNestHost returns an error code if setting the
+				 * nest host field fails.
+				 *
+				 * Note that the specification asserts an order for nest host
+				 * loading during access checking. The accessing class's nest
+				 * host is loaded first and the class being accessed loads its
+				 * nest host after.
+				 */
+				if (NULL == sourceClass->nestHost) {
+					result = loadAndVerifyNestHost(currentThread, sourceClass, lookupOptions);
+				}
+				if (J9_VISIBILITY_ALLOWED != result) {
```

Comment:
why isn't this in the same scope has the `(NULL == sourceClass->nestHost)` check above